server/views/pages/createGoals/index.njk

```diff
@@ -43,7 +43,7 @@
           <div class="govuk-summary-card__content">
 
             {{ govukCharacterCount({
-              name: "goals[" + goalLoop.index0 + "].title",
+              name: "goals[" + goalLoop.index0 + "][title]",
```

Comment:
Adam got a bit confused with the HTML field name syntax vs. the `id` attribute syntax, bless him 😁 
He got especially confused with the syntax for the steps (below), though to be fair that is messy!

So, our form definition / object graph is this:
```
  export interface CreateGoalsForm {
    prisonNumber: string
    goals: Array<{
      title?: string
      targetCompletionDate?: string
      'targetCompletionDate-day'?: string
      'targetCompletionDate-month'?: string
      'targetCompletionDate-year'?: string
      steps: Array<{
        title?: string
      }>
      note?: string
    }>
    action?: 'add-another-step' | 'submit-form'
  }
```
eg:
```
{
  prisonNumber: 'A1234BC',
  goals: [
    {
      title: 'Goal 1',
      targetCompletionDate: '2024-12-31',
      steps: [
        { title: 'Step 1 for Goal 1' },
        { title: 'Step 2 for Goal 1' },
      ],
      note: 'A note about goal 1'
    },
    {
      title: 'Goal 2',
      targetCompletionDate: '2025-02-28',
      steps: [
        { title: 'Step 1 for Goal 2' },
        { title: 'Step 2 for Goal 2' },
      ],
      note: 'A note about goal 2'
    },
  ],
  action: 'submit-form',
}
```

The html form field name for array indexing needs to be quite specific for the express urlencoded body-parser to bind it into the object graph properly. Unfortunately it doesnt do dot notation! 😢 

To index into a goal and reference it's `title` field (which is a field directly off of a goal array element) it needs to be:
```
goals[0][title]
```

To index into a step within a goal and reference it's `title` field (which is a field directly off of a step array element) it needs to be:
```
goals[0][steps][0][title]
```

For the `id` element we can use whatever we like (as long as it's a valid & unique ID) - dot notation is nice as we are all familiar with it and it reflects the object graph well